myriad-scheduler/src/main/java/com/ebay/myriad/scheduler/event/handlers/ResourceOffersEventHandler.java

```diff
@@ -74,20 +76,35 @@ public void onEvent(ResourceOffersEvent event, long sequence,
     SchedulerDriver driver = event.getDriver();
     List<Offer> offers = event.getOffers();
 
+    // Sometimes, we see that mesos sends resource offers before Myriad receives
+    // a notification for "framework registration". This is a simple defensive code
+    // to not process any offers unless Myriad receives a "framework registered" notification.
+    if (schedulerState.getFrameworkID() == null) {
+      LOGGER.warn("Received {} offers, but declining them since Framework ID is not yet set", offers.size());
+      for (Offer offer : offers) {
+        driver.declineOffer(offer.getId());
+      }
+      return;
+    }
     LOGGER.info("Received offers {}", offers.size());
     LOGGER.debug("Pending tasks: {}", this.schedulerState.getPendingTaskIds());
     driverOperationLock.lock();
     try {
       for (Iterator<Offer> iterator = offers.iterator(); iterator.hasNext();) {
         Offer offer = iterator.next();
+        NodeTask nodeTask = schedulerState.getNodeTask(offer.getSlaveId());
```

Comment:
Although I don't think anybody's currently working on dynamic slave attributes, it's definitely on the roadmap.
See https://issues.apache.org/jira/browse/MESOS-1739 and https://issues.apache.org/jira/browse/MESOS-3059
When it does happen, Mesos will likely have to notify frameworks with running tasks on the reconfigured node that the attributes have changed, in case the framework would want to cancel/reschedule given the new knowledge.

Imagine a node that does not have the FUBAR attribute when we originally launch a Myriad NM task on it, but later gets tagged as FUBAR. What should the expected behavior be if we ask to flex-down a NM on a FUBAR node? Do we flex down the task because it is on a FUBAR node, or do we leave this task running, since the node was not FUBAR when we launched the task?